### ISSUE TITLE
Badge for merged pull requests.

### ISSUE THREAD

**Issue by AraHaan** [badges/shields#6164]
:clipboard: **Description**

I would like a badge that counts all of the pull requests that was merged only, the closed pr badge could be changed I think to filter out merged prs.

- The badge is for GitHub.
- What sort of information should this badge show?
The badge should contain the count of merged pull requests, also the count of merged pull requests could also be used to subtract from the closed pull request badge to (or well make a version of it that excludes merged prs).

:link: [![example](https://img.shields.io/badge/merged%20pull%20requests-100%20merged-darkviolet)]()

Where can we get the data from?
``GET https://api.github.com/repos/{owner}/{repo}/pulls/{pull_number}/merge``

- Is there a public API?
yes
- Does the API requires an API key?
Nope.

- Link to the API documentation.
https://docs.github.com/en/rest/reference/pulls#check-if-a-pull-request-has-been-merged

:microphone: **Motivation**

I noticed that we could show up opened, closed issues and pull requests among other things however lacked the ability to get a badge for merged pull requests.

- What is the specific use case?
People like me want to show that we merge pull requests a ton and it would support those who feel like contributing to projects too.

**Comment by AraHaan** [comment]
It seems there is another way as well to -> https://github.com/Elskom/Sdk/pulls?q=is%3Apr+is%3Aclosed+is%3Aunmerged
returns only unmerged pr's while -> https://github.com/Elskom/Sdk/pulls?q=is%3Apr+is%3Aclosed+is%3Amerged
returns only merged.

**Comment by calebcartwright** [comment]
Thanks! I think that the API information is not correct for what you're asking for though, or at least not a viable option.

We actually already support distinguishing between merged vs. closed for _individual_ PRs

https://github.com/badges/shields/pull/6159
https://img.shields.io/github/pulls/detail/state/badges/shields/6159
![](https://img.shields.io/github/pulls/detail/state/badges/shields/6159)

https://github.com/badges/shields/pull/6161
https://img.shields.io/github/pulls/detail/state/badges/shields/6161
![](https://img.shields.io/github/pulls/detail/state/badges/shields/6161)

The difference here is the desire to filter against the total PR count for merged vs. closed, and this is actually much simpler to do via the v4/GraphQL API. Fortunately, our existing badge that reports total for both issues and PRs already does this!

https://github.com/badges/shields/blob/d4116b3ab0bb05fdad5695b5f884b38b6f4af954/services/github/github-issues.service.js#L332-L349

Think all that would be needed to make this happen is an additional route variant (`issues-pr-merged`, for consistency)
https://github.com/badges/shields/blob/d4116b3ab0bb05fdad5695b5f884b38b6f4af954/services/github/github-issues.service.js#L42-L46

and then tweaking the respective signatures to support passing through the merged-only option, and using the state filters accordingly:
https://github.com/badges/shields/blob/d4116b3ab0bb05fdad5695b5f884b38b6f4af954/services/github/github-issues.service.js#L345-L348

closed = merged+closed states
merged = merged state
otherwise open state

**Comment by AraHaan** [comment]
@calebcartwright can we also do an ``issues-pr-closed-unmerged`` too?

**Comment by calebcartwright** [comment]
> `issues-pr-closed-unmerged` too eventually?

would probably need to bikeshed on the name a bit, but i don't see why not!

**Comment by chris48s** [comment]
> It seems there is another way as well to -> https://github.com/Elskom/Sdk/pulls?q=is%3Apr+is%3Aclosed+is%3Aunmerged
> returns only unmerged pr's while -> https://github.com/Elskom/Sdk/pulls?q=is%3Apr+is%3Aclosed+is%3Amerged
> returns only merged.

You can use these queries with the custom issue search badge e.g:

![](https://img.shields.io/github/issues-search/Elskom/Sdk?label=unmerged%20PRs&query=is%3Apr%20is%3Aclosed%20is%3Aunmerged) - https://img.shields.io/github/issues-search/Elskom/Sdk?label=unmerged%20PRs&query=is%3Apr%20is%3Aclosed%20is%3Aunmerged

![](https://img.shields.io/github/issues-search/Elskom/Sdk?label=merged%20PRs&query=is%3Apr+is%3Aclosed+is%3Amerged) - https://img.shields.io/github/issues-search/Elskom/Sdk?label=merged%20PRs&query=is%3Apr+is%3Aclosed+is%3Amerged

**Comment by calebcartwright** [comment]
Good point Chris. Do you think we should go ahead and close this (folks can always use the `color` query param if they want it to have the same purple color) or would it be worthwhile to make those options available like the other PR aggregate status badges?

**Comment by AraHaan** [comment]
I would like the options available like the other pr aggregate badges.

**Comment by AraHaan** [comment]
https://github.com/badges/shields/blob/master/services/github/github-issues-search.service.js#L26-L28

I see from here it does not let you set text at the end.

**Comment by calebcartwright** [comment]
> I see from here it does not let you set text at the end, perhaps have an optional query param for it that defaults to nothing?.

No, this is by design. We do not allow open ended badge message customization #6135. For the issues status badges, we have two kinds, one that puts the status in the label (raw) and another that puts it as a suffix in the message:

https://img.shields.io/github/issues-pr-raw/badges/shields
![](https://img.shields.io/github/issues-pr-raw/badges/shields)

https://img.shields.io/github/issues-pr/badges/shields 
![](https://img.shields.io/github/issues-pr/badges/shields)

the former is already possible via the issue search badge and query params as noted above, the latter would not be.

i'm not personally opposed to to including the additional PR status variants, but it does seem like quite a bit of overhead just to get the non-raw versions (status on the message)

**Comment by jaas666** [comment]
This issue seems like it was solved. 
Can I suggest closing it?

**Comment by calebcartwright** [comment]
> This issue seems like it was solved. Can I suggest closing it?

Thanks for the question! However, no, I don't see this as resolved.

It's often the case that there's a viable alternative/workaround available via existing features such as our dynamic badge features (or custom endpoint), but that doesn't inherently negate/resolve the request for a native badge.

This issue was requesting a native badge that doesn't yet exist (though which has some alternative as noted earlier in the thread). From the maintainer side we've not explicitly said "no we won't do this", and the author hasn't changed their mind yet either and closed the issue themselves.

I don't anticipate the issue being closed unless and until one of those two things happen, or unless someone implements the requested badge.

**Comment by jaas666** [comment]
Noted 👍

**Comment by PaulaBarszcz** [comment]
I would like to work on this one :)
